packages/scxa-marker-gene-heatmap/src/MarkerGeneHeatmap.js

```diff
@@ -102,11 +102,14 @@ const MarkerGeneHeatmap = (props) => {
         value: plotLineAxisPosition,
         zIndex,
         label: heatmapType !== `multiexperimentcelltypes` && {
-          text: splitCellTypeLabel.join(`<br/>`),
+          // tilted and trim very long labels for long cell types
+          text: splitCellTypeLabel.length <= 3 ? splitCellTypeLabel.join(`<br/>`) : splitCellTypeLabel.slice(0, 3).join(`<br/>`) + `...`,
           align: `right`,
           textAlign: `left`,
-          x: 0,
-          y: yOffset - Math.max(splitCellTypeLabel.length - 3, 0) * heatmapRowHeight / 2 // Move very long labels slightly up
+          rotation: heatmapType === `celltypes` ? -45 : 0,
+          x: heatmapType === `celltypes` ? 3 : 0,
+          // No need to move very long labels slightly up as we tilted and trim it for long cell types
```

Comment:
Could you remove this comment, please?
Having less or no comment in the code is always better as the comments unfortunately getting rotten as time goes.
If somebody change the code later, but leave the comment here, then it is going to confuse a new developer. It is better to leave less opportunity for this happening.
The best comment is always a non-existent comment. ;-)
We are using version control, so you can add this comment to your commit in the description session of the commit message. It is a much better place for this kind of comments.
How to do it: https://stackoverflow.com/questions/16122234/how-to-commit-a-change-with-both-message-and-description-from-the-command-li/36427485#36427485
Thanks